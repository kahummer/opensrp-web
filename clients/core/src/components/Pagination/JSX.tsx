/* eslint-disable react/prop-types */
import * as React from 'react';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';
import { PaginationState } from '@onaio/pagination';
import { ExtendingOptions } from '.';
import { Action } from 'history';

interface ActionCreators {
    (): void;
}

interface Props {
    paginationState: PaginationState<ExtendingOptions>;
    nextPage: ActionCreators;
    firstPage: ActionCreators;
    lastPage: ActionCreators;
    goToPage: (page: number) => void;
    previousPage: ActionCreators;
    canNextPage: boolean;
    canPreviousPage: boolean;
    fetchMoreApiData: () => Promise<void>;
}

/** JSX only Wrapper for bootstrap pagination component*/
export const BootstrapJSX: React.FC<Props> = props => {
    const {
        paginationState,
        nextPage,
        firstPage,
        lastPage,
        goToPage,
        previousPage,
        canNextPage,
        canPreviousPage,
        fetchMoreApiData,
    } = props;

    return (
        <React.Fragment>
            <Pagination aria-label="pagination" size="sm">
                <PaginationItem className={`page-item ${canPreviousPage ? '' : 'disabled'}`}>
                    <PaginationLink
                        className={`page-link`}
                        href="#"
                        aria-label="Start"
                        // tslint:disable-next-line: jsx-no-lambda
                        onClick={() => firstPage()}
                    >
                        <span aria-hidden="true">Start</span>
                        <span className="sr-only">Start</span>
                    </PaginationLink>
                </PaginationItem>
                <PaginationItem className={`page-item ${canPreviousPage ? '' : 'disabled'}`}>
                    <PaginationLink
                        className={`page-link`}
                        href="#"
                        aria-label="Previous"
                        // tslint:disable-next-line: jsx-no-lambda
                        onClick={() => previousPage()}
                    >
                        <span aria-hidden="true">Previous</span>
                        <span className="sr-only">Previous</span>
                    </PaginationLink>
                </PaginationItem>
                {paginationState.pagesToDisplay.map((page: any, index: number) => {
                    return (
                        <PaginationItem
                            key={index}
                            className={`page-item ${paginationState.currentPage === page ? ' active' : ''}`}
                        >
                            {/* tslint:disable-next-line:jsx-no-lambda */}
                            <PaginationLink className="page-link" href="#" onClick={() => goToPage(page)}>
                                {page}
                            </PaginationLink>
                        </PaginationItem>
                    );
                })}
                {paginationState.canShowEllipsis && (
                    <PaginationItem className={`page-item`}>
                        <PaginationLink
                            className={`page-link`}
                            href="#"
                            aria-label={'Ellipsis'}
                            // tslint:disable-next-line: jsx-no-lambda
                            onClick={() => fetchMoreApiData()}
                        >
                            <span aria-hidden="true">ellipsis</span>
                            <span className="sr-only"> {paginationState.ellipsisIsLoading ? 'Loading' : '...'} </span>
                        </PaginationLink>
                    </PaginationItem>
                )}
                <PaginationItem className={`page-item  ${canNextPage ? '' : 'disabled'}`}>
                    <PaginationLink
                        className={`page-link`}
                        href="#"
                        aria-label={'Next'}
                        // tslint:disable-next-line: jsx-no-lambda
                        onClick={() => nextPage()}
                    >
                        <span aria-hidden="true">Next</span>
                        <span className="sr-only">Next</span>
                    </PaginationLink>
                </PaginationItem>
                <PaginationItem className={`page-item ${canNextPage ? '' : 'disabled'}`}>
                    <PaginationLink
                        href="#"
                        aria-label="Last"
                        // tslint:disable-next-line: jsx-no-lambda
                        onClick={() => lastPage()}
                    >
                        <span aria-hidden="true">Last</span>
                        <span className="sr-only">Last</span>
                    </PaginationLink>
                </PaginationItem>
            </Pagination>
        </React.Fragment>
    );
};
