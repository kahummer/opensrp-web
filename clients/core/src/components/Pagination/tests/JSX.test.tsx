import React from 'react';
import { shallow, mount } from 'enzyme';
import { BootstrapJSX } from '../JSX';

describe('src/components/pagination/bootsrapJSX', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    xit('renders without crashing', () => {
        const genericMock = jest.fn();
        const props = {
            paginationState: {
                ellipsisIsLoading: false,
                showEndingEllipsis: false,
                currentPage: 4,
                pagesToDisplay: ['2', '3', '4', '5', '6'],
            },
            nextPage: genericMock,
            firstPage: genericMock,
            lastPage: genericMock,
            goToPage: genericMock,
            previousPage: genericMock,
            canNextPage: true,
            canPreviousPage: true,
            fetchMoreApiData: genericMock,
        };
        shallow(<BootstrapJSX {...props} />);
    });

    xit('renders correctly', () => {
        const genericMock = jest.fn();
        const props = {
            paginationState: {
                ellipsisIsLoading: false,
                showEndingEllipsis: false,
                currentPage: 4,
                pagesToDisplay: ['2', '3', '4', '5', '6'],
            },
            nextPage: genericMock,
            firstPage: genericMock,
            lastPage: genericMock,
            goToPage: genericMock,
            previousPage: genericMock,
            canNextPage: true,
            canPreviousPage: true,
            fetchMoreApiData: genericMock,
        };
        const wrapper = mount(<BootstrapJSX {...props} />);
        expect(wrapper.text()).toEqual('');
    });

    xit('pagination items invokes the correct callbacks', () => {
        const genericMock = jest.fn();
        const props = {
            paginationState: {
                ellipsisIsLoading: false,
                showEndingEllipsis: false,
                currentPage: 4,
            },
            nextPage: genericMock,
            firstPage: genericMock,
            lastPage: genericMock,
            goToPage: genericMock,
            previousPage: genericMock,
            canNextPage: true,
            canPreviousPage: true,
            fetchMoreApiData: genericMock,
        };
        shallow(<BootstrapJSX {...props} />);
    });
});
