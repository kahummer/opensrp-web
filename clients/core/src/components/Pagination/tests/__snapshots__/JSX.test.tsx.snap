// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/pagination/bootsrapJSX ellipsis operations work correctly : ellipsis pagination  link when loading 1`] = `
<PaginationItem
  className="page-item"
  tag="li"
>
  <li
    className="page-item page-item"
  >
    Loading
  </li>
</PaginationItem>
`;

exports[`src/components/pagination/bootsrapJSX ellipsis operations work correctly when not loading : ellipsis pagination link 1`] = `
<PaginationItem
  className="page-item"
  tag="li"
>
  <li
    className="page-item page-item"
  >
    <PaginationLink
      aria-label="ellipsis"
      className="page-link"
      href="#"
      onClick={[Function]}
      tag="a"
    >
      <a
        aria-label="ellipsis"
        className="page-link page-link"
        href="#"
        onClick={[Function]}
      >
        . . .
      </a>
    </PaginationLink>
  </li>
</PaginationItem>
`;
